packages/backend/helpers/rateLimitMiddleware.ts

```diff
@@ -0,0 +1,54 @@
+import { Request, Response } from 'express';
+import { RateLimiterRedis, IRateLimiterStoreOptions } from 'rate-limiter-flexible';
+
+import redis from '../redis/client';
+import { skipRateLimitRoutes } from './utils';
+
+//We can make this dynamic based on the subscription as well
+const RATE_LIMIT_DURATION_IN_MINUTES = 1;
+//TODO:Should we keep DEFAULT_RATE_LIMIT_DEVELOPER_PLAN value in env variable?
+const DEFAULT_RATE_LIMIT_DEVELOPER_PLAN = 100;
+
+class RateLimiterManager {
+    // In Memory Cache for storing RateLimiterRedis instances to prevent the creation of a new instance for each request.
```

Comment:
@shadab14meb346 can't a singleton/single instance of the the RateLimiter not work for us? 

Since we manage keys & their consumption limits by the unique key anyways, I'm not sure if I follow if this is required.

Reply:
You are right about the singleton instance works for us here.
The only reason to use class here is to encapusulate the `#rateLimiters` so by mistake no one can export it is we keep it as a global variable in the file. Earlier it was something like [this](https://github.com/shadab14meb346/revert/blob/6e571068a8d07a1bbc0d3a4765debc884f002c8b/packages/backend/helpers/rateLimitMiddleware.ts#L12)

Reply:
@shadab14meb346 the previous implementation seems better to me tbh.